pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,348 @@
+	reqLogger.Info("Reconciling Istio")
+
+	// Get the Istio CR.
+	istio, msg, err := GetIstio(ctx, r)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info("Istio object not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying for Istio CR: "+msg, err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	r.status.OnCRFound()
+
+	// SetMetaData in the TigeraStatus such as observedGenerations.
+	defer r.status.SetMetaData(&istio.ObjectMeta)
+
+	// Get the Installation, for private registry and pull secret config.
+	variant, installation, err := utils.GetInstallation(ctx, r)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if variant == "" {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Installation Variant to be set", nil, reqLogger)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// patch felixconfiguration
+	_, err = utils.PatchFelixConfiguration(ctx, r.Client, func(fc *crdv1.FelixConfiguration) (bool, error) {
+		var istioMode *string
+		if fc.Annotations[render.IstioOperatorAnnotationMode] != "" {
+			value := fc.Annotations[render.IstioOperatorAnnotationMode]
+			istioMode = &value
+		}
+
+		// the variables only can be equal if they are nil
+		if istioMode != fc.Spec.IstioMode &&
+			(istioMode == nil || fc.Spec.IstioMode == nil || *istioMode != *fc.Spec.IstioMode) {
+			return false, fmt.Errorf("felixconfig %q modified by user", "IstioMode")
+		}
+
+		istioModeDesired := "Enabled"
+		fc.Spec.IstioMode = &istioModeDesired
+		if fc.Annotations == nil {
+			fc.Annotations = make(map[string]string)
+		}
+		fc.Annotations[render.IstioOperatorAnnotationMode] = istioModeDesired
+
+		if istio.Spec.DSCPValue == nil {
+			return true, nil
+		}
+
+		var dscpValue *uint32
+		if fc.Annotations[render.IstioOperatorAnnotationDSCP] != "" {
+			value, err := strconv.ParseUint(fc.Annotations[render.IstioOperatorAnnotationDSCP], 10, 32)
+			if err != nil {
+				return false, err
+			}
+			u32Value := uint32(value)
+			dscpValue = &u32Value
+		}
+
+		// the variables only can be equal if they are nil
+		if dscpValue != fc.Spec.IstioDSCPValue &&
+			(dscpValue == nil || fc.Spec.IstioDSCPValue == nil || *dscpValue != *fc.Spec.IstioDSCPValue) {
+			return false, fmt.Errorf("felixconfig %q modified by user", "IstioDSCPValue")
+		}
+
+		fc.Spec.IstioDSCPValue = istio.Spec.DSCPValue
+		fc.Annotations[render.IstioOperatorAnnotationMode] = strconv.FormatUint(uint64(*istio.Spec.DSCPValue), 32)
+
+		return true, nil
+	})
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error rendering essential Tigera Istio CRDs", err, log)
```

Comment:
```suggestion
		r.status.SetDegraded(operatorv1.ResourceCreateError, "Error patching felix configuration with Istio settings", err, log)
```

The more specific we can make our error messages the better. "Essential Tigera Istio CRDs" will be very vague when attempting to diagnose error logs.